File: prime_factorisation.php
<?php
function primeFactorisation($n) {
    $factors = array();

    while ($n % 2 == 0) {
        $factors[] = 2;
        $n = $n / 2;
    }

    for ($i = 3; $i <= sqrt($n); $i = $i + 2) {
        while ($n % $i == 0) {
            $factors[] = $i;
            $n = $n / $i;
        }
    }

    if ($n > 2) {
        $factors[] = $n;
    }

    return $factors;
}
?>

File: prime_factorisation.m
function factors = PrimeFactorisation(n)
    factors = [];
    while mod(n, 2) == 0
        factors = [factors, 2];
        n = n / 2;
    end
    for i = 3:sqrt(n)
        while mod(n, i) == 0
            factors = [factors, i];
            n = n / i;
        end
    end    
    if n > 2
        factors = [factors, n];
    end
end
